depthai_ros_driver/CMakeLists.txt

```diff
@@ -1,171 +1,81 @@
-RETURN()
+  )
 else()
-  message(VERBOSE "ROS version is $ENV{ROS_VERSION}, skipping ROS1 ${PROJECT_NAME}")
+  set (PKG_DEPS
+    ${PKG_DEPS}
+    rclcpp
+    rclcpp_components
+  )
 endif()
 
+include(cmake/ros_version_agnostic_add_node.cmake)
+
+# A bug in the CMake of the included external project depthai-core clobbers the
+# paths used to find ROS packages, so we have to search for them first and then
+# add the external project
+find_dependent_packages(${PKG_DEPS})
+
+add_subdirectory(${DEPTHAI_CORE} depthai-core)
+
+add_ros_node(depthai_ros_driver
+  SKIP_PACKAGE_SEARCH
+  PROJECT_NAME ${PROJECT_NAME}
+  COMPONENT_CLASS "rr::DepthAIBaseRos2"
+  SOURCES src/depthai_common.cpp
+  ROS1_SOURCES src/depthai_base.cpp
+  ROS2_SOURCES src/depthai_base_ros2.cpp
+  ROS1_COMPONENT_SOURCES src/depthai_nodelet.cpp
+  ROS1_EXE_SOURCES src/depthai_node.cpp
+  ROS2_EXE_SOURCES src/depthai_node_ros2.cpp
+  DIRECTORIES launch resources scripts urdf params
+  INCLUDE_DIRS include
+  SYSTEM_DEPS OpenCV Boost
+  PKG_DEPS ${PKG_DEPS}
+  TARGET_DEPS depthai-core)
 
-#==============================================================================
-# Compile in ROS1
-#==============================================================================
-
-if($ENV{ROS_VERSION} STREQUAL 2)
-    find_package(ament_cmake REQUIRED)
-    set (PKG_DEPS
-        ${PKG_DEPS_COMMON}
-        rclcpp
-        rclcpp_components
-    )
-    foreach(pkg ${PKG_DEPS})
-        find_package(${pkg} REQUIRED)
-    endforeach()
-
-    # Add depthai-core dependency
-    add_subdirectory (${DEPTHAI_CORE} depthai-core)
-
-    add_definitions(-DUSE_ROS2)   # <-------- Set to use ros2 macro, for #ifdef
-    add_library (${PROJECT_NAME}_lib SHARED
-        src/depthai_common.cpp
-        src/depthai_base_ros2.cpp
-    )
-    target_link_libraries (${PROJECT_NAME}_lib PUBLIC
-        ${OpenCV_LIBRARIES}
-        ${Boost_LIBRARIES}
-        depthai-core
-    )
-    ament_target_dependencies(${PROJECT_NAME}_lib 
-        PUBLIC ${PKG_DEPS}
-    )
-
-    # Generate ros2 component
-    rclcpp_components_register_nodes(${PROJECT_NAME}_lib  "rr::DepthAIBaseRos2")
-    
-    include_directories (include)
-    add_executable (${PROJECT_NAME}_node src/depthai_node_ros2.cpp)
-    target_link_libraries(${PROJECT_NAME}_node PUBLIC ${PROJECT_NAME}_lib)
-
-    # Install ros2 component
-    install(TARGETS
-        ${PROJECT_NAME}_lib
-        ARCHIVE DESTINATION lib
-        LIBRARY DESTINATION lib
-        RUNTIME DESTINATION bin)
-
-    install(DIRECTORY
-        launch/ params resources urdf
-        DESTINATION share/${PROJECT_NAME}
-    )
-
-    install(TARGETS
-        ${PROJECT_NAME}_node
-        DESTINATION lib/${PROJECT_NAME})
-
-    ament_package()
-else()
-    message(VERBOSE "ROS version is $ENV{ROS_VERSION}, skipping this ROS2 ${PROJECT_NAME}")
+if($ENV{ROS_VERSION} STREQUAL 1)
+  install(FILES nodelet_plugins.xml DESTINATION ${depthai_ros_driver_PACKAGE_SHARE_DESTINATION})
```

Comment:
Is it possible to move this into the macro? (let `DIRECTORIES` accepts files??) I guess there's no clean way to do it too.